fire/SFMainMod.F90

```diff
@@ -322,19 +337,180 @@ subroutine  charecteristics_of_fuel ( currentSite )
+  !****************************************************************.  
+
+    !returns the live crown fuel characteristics within each patch.
+    ! passive_crown_FI is minimum fire intensity to ignite canopy crown fuel
+
+    use SFParamsMod,    only : SF_VAL_CWD_FRAC
+
+    type(ed_site_type), intent(in), target :: currentSite
+
+    type(ed_patch_type) , pointer :: currentPatch
+    type(ed_cohort_type), pointer :: currentCohort
+
+    ! ARGUMENTS
+    real(r8), intent(out) :: canopy_fuel_load     ! available canopy fuel load in patch (kg biomass)
+    real(r8), intent(out) :: passive_crown_FI     ! min fire intensity to ignite canopy fuel (kW/m)
+
+    ! LOCAL
+    real(r8) ::  crown_depth          ! depth of crown (m)
+    real(r8) ::  height_cbb           ! clear branch bole height or crown base height (m)
+    real(r8) ::  max_height           ! max cohort on patch (m)
+    real(r8) ::  crown_ignite_energy  ! heat yield for crown (kJ/kg)
+    real(r8) ::  tree_sapw_struct_c   ! above-ground tree struct and sap biomass in cohort (kgC)
+    real(r8) ::  leaf_c                  ! leaf carbon (kgC)
+    real(r8) ::  sapw_c                  ! sapwood carbon (kgC)
+    real(r8) ::  struct_c                ! structure carbon (kgC)
+    real(r8) ::  twig_sapw_struct_c      ! above-ground twig sap and struct in cohort (kgC)
+    real(r8) ::  crown_fuel_c            ! biomass of 1 hr fuels (leaves,twigs) in cohort (kg C)
+    real(r8) ::  crown_fuel_biomass      ! biomass of crown fuel in cohort (kg biomass)
+    real(r8) ::  crown_fuel_per_m        ! crown fuel per 1m section in cohort
+    real(r8) ::  height_base_canopy      ! lowest height of fuels in patch to carry fire in crown
+    real(r8) ::  canopy_bulk_density     ! density of canopy fuel on patch
+
+    integer  ::  ih                      ! counter
+
+    real, dimension(70):: biom_matrix   ! matrix to track biomass from bottom to 70m
+    real(r8),parameter :: min_density_canopy_fuel = 0.011_r8 !min canopy fuel density (kg/m3) sufficient to
+                                                             !propogate fire vertically through canopy
+                                                             !Scott and Reinhardt 2001 RMRS-RP-29
+    real(r8),parameter :: foliar_moist_content = 1.0_r8      !foliar moisture content default 100% Scott & Reinhardt 2001
+
+
+    !returns the live crown fuel characteristics within each patch.
+    ! passive_crown_FI is the required minimum fire intensity to ignite canopy crown fuel
+
+    currentPatch => currentSite%oldest_patch
+
+    !! check to see if active_crown_fire is enabled? 
+
+    do while(associated(currentPatch))
+       !zero Patch level variables
+       height_base_canopy                   = 0.0_r8
+       canopy_fuel_load                     = 0.0_r8
+       passive_crown_FI                     = 0.0_r8
+       max_height = 0._r8
+       biom_matrix(:) = 0._r8
+
+!      if (currentPatch%active_crown_fire == 1) then
```

Comment:
@jkshuman I see your comment here. I'm not putting the if-statement here because it will prevent `passive_crown_FI` from becoming greater than 0 when we run without active crown fire.

I am adding the if-statement later, after the comment
`! check threshold intensity and rate of spread`
That's the only place where I think this if-statement is required.